adr/ADR-22.md

```diff
@@ -0,0 +1,184 @@
+# Client File Locations
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2022-01-24|
+|Author  |@philpennock|
+|Status  |Proposed|
+|Tags    |client, config|
+
+
+## Context and Problem Statement
+
+The configuration files for NATS clients should be in a coherent predictable location.
+Clients in various languages should be able to know where to read and write data,
+and administrators should know what file-system access is needed and where
+sensitive key data might be stored.
+
+This document both unifies the location and acts as a living registry of the
+namespace within the NATS configuration directory.
+
+
+## Version History
+
+|Date      |Revision|
+|----------|--------|
+|2022-01-24|Initial proposal|
+
+
+## [Context | References | Prior Work]
+
+* [ADR-21][] on Configuration Contexts moves us closer to the model
+  described herein, but ADR-21 will need updating for the non-XDG platforms.
+* The `nsc` and `ngs` commands use other locations, scattering files around.
```

Comment:
`ngs` tool doesn't write any state file (pretty sure)
`nats` cli is the only other cli tool